Gems/ExpressionEvaluation/Code/Source/ElementInformationSerializer.inl

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Memory/Memory.h>
+#include <AzCore/Serialization/AZStdAnyDataContainer.inl>
+#include <AzCore/Serialization/Json/BaseJsonSerializer.h>
+#include <AzCore/Serialization/Json/JsonSerialization.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <ExpressionEvaluation/ExpressionEngine/ExpressionTypes.h>
+
+namespace AZ
+{
+    class ElementInformationSerializer
+        : public BaseJsonSerializer
+    {
+
+    public:
+        AZ_RTTI(ElementInformationSerializer, "{B33E6AA9-C700-4E3D-857C-55F362AFE57A}", BaseJsonSerializer);
+        AZ_CLASS_ALLOCATOR_DECL;
+
+    private:
+        using ElementInformation = ExpressionEvaluation::ElementInformation;
+
+        static const char* EmptyAnyIdentifier;
+
+        static bool IsEmptyAny(const rapidjson::Value& typeId)
+        {
+            if (typeId.IsString())
+            {
+                AZStd::string_view typeName(typeId.GetString(), typeId.GetStringLength());
+                return typeName == EmptyAnyIdentifier;
+            }
+
+            return false;
+        }
+
+        JsonSerializationResult::Result Load
+            ( void* outputValue
+            , const Uuid& outputValueTypeId
+            , const rapidjson::Value& inputValue
+            , JsonDeserializerContext& context) override
+        {
+            namespace JSR = JsonSerializationResult;
+
+            AZ_Assert(outputValueTypeId == azrtti_typeid<ElementInformation>(), "ElementInformationSerializer Load against "
+                "output typeID that was not ElementInformation");
+            AZ_Assert(outputValue, "ElementInformationSerializer Load against null output");
+
+            JsonSerializationResult::ResultCode result(JSR::Tasks::ReadField);
+            auto outputDatum = reinterpret_cast<ElementInformation*>(outputValue);
+            result.Combine(ContinueLoadingFromJsonObjectField
+                ( &outputDatum->m_id
+                , azrtti_typeid<decltype(outputDatum->m_id)>()
+                , inputValue
+                , "Id"
+                , context));
+
+            // any storage begin
+            {
+                AZ::Uuid typeId = AZ::Uuid::CreateNull();
+                auto typeIdMember = inputValue.FindMember(JsonSerialization::TypeIdFieldIdentifier);
+                if (typeIdMember == inputValue.MemberEnd())
+                {
+                    return context.Report
+                        ( JSR::Tasks::ReadField
+                        , JSR::Outcomes::Missing
+                        , AZStd::string::format("ElementInformationSerializer::Load failed to load the %s member"
+                            , JsonSerialization::TypeIdFieldIdentifier));
```

Comment:
Repeated: This wouldn't necessarily be an issue. If default stripping was enabled then it would mean that the target type is the same as the type that's already used by the target.